### ISSUE TITLE
cli: Migrate status command got very slow after update to 2.5.0

### ISSUE THREAD

**Issue by vaughanj10** [hasura/graphql-engine#8398]
### Version Information

Server Version:
CLI Version (for CLI related issue): v2.5.0

### Environment

OSS

### What is the expected behaviour?

The expected behaviour is that when I run: `hasura migrate status --database-name default`, I expect this command to return the migration status promptly (< 1 second). It historically has been very fast until I updated to 2.5.0 today. 

### Keywords

cli, slow, hasura migrate status, 2.5.0

### What is the current behaviour?

The current behavior is that when I run: `hasura migrate status --database-name default`, it takes > 30 seconds. 

### How to reproduce the issue?

1. Update the Hasura CLI to v2.5.0 
2. Run `hasura migrate status --database-name default` 

```

❯ sudo hasura update-cli --version v2.3                                                  
INFO Updated to latest version                     version=v2.3.0


❯ time hasura migrate status --database-name default --skip-update-check > /dev/null 2>&1
hasura migrate status --database-name default --skip-update-check > /dev/null  0.98s user 0.10s system 85% cpu 1.268 total


❯ sudo hasura update-cli --version v2.4                                                  
INFO Updated to latest version                     version=v2.4.0


❯ time hasura migrate status --database-name default --skip-update-check > /dev/null 2>&1
hasura migrate status --database-name default --skip-update-check > /dev/null  0.98s user 0.07s system 75% cpu 1.387 total


❯ sudo hasura update-cli --version v2.5                                                  
INFO Updated to latest version                     version=v2.5.0


❯ time hasura migrate status --database-name default --skip-update-check > /dev/null 2>&1
hasura migrate status --database-name default --skip-update-check > /dev/null  108.64s user 3.33s system 277% cpu 40.331 total

```

### Please provide any traces or logs that could help here.

```
❯ hasura migrate status --database-name default --skip-update-check --log-level DEBUG           
DEBU global config file exists, verifying contents 
DEBU global config is pre-set to &cli.GlobalConfig{UUID:"3e172b6e-0435-4509-946a-9707b2617e99", EnableTelemetry:true, ShowUpdateNotification:true, CLIEnvironment:"default"} 
DEBU global config: uuid: 3e172b6e-0435-4509-946a-9707b2617e99 
DEBU global config: enableTelemetry: true         
DEBU global config: showUpdateNotification: true  
DEBU global config: cliEnvironment: default       
DEBU versions: cli: [v2.5.0] server: [v2.5.0]     
DEBU compatibility check: [true] compatible CLI and Server 
DEBU cannot get config information from server, this might be because config API is not enabled: error unmarshalling fetching server config 
DEBU graphql engine endpoint: http://localhost:8080 
DEBU graphql engine admin_secret: ‹×›             
DEBU server: uuid: 5099d1b2-556a-4680-a4ee-9635185bf9d9 
DEBU skipping state migration, found IsStateCopyCompleted: true Migrations: map[default:map[1629338136826:false 1629566119947:false 1629566373573:false 1629685818000:false 1629765007782:false 1629766111544:false 1629767142747:false 1629781120490:false 1629781138162:false 1629781157801:false 1629781830053:false 1629819147273:false 1629820358459:false 1629921396032:false 1630078206540:false 1630078506487:false 1630166646615:false 1630172959829:false 1630350966389:false 1630442100598:false 1630518845858:false 1630855428046:false 1630926634938:false 1630927653550:false 1630972638684:false 1631069451749:false 1631116885160:false 1631117860352:false 1631135875193:false 1631208373636:false 1631220292189:false 1631230740775:false 1631297215368:false 1631309515904:false 1631378806389:false 1631460881218:false 1631461276368:false 1631627395072:false 1631708179158:false 1631708213784:false 1631708869032:false 1631709222555:false 1631739531156:false 1631798471257:false 1631827810077:false 1631903064289:false 1631934250321:false 1631935257366:false 1632002118092:false 1632002623142:false 1632225306336:false 1632225340725:false 1632231486912:false 1632241371186:false 1632262172947:false 1632273185941:false 1632273482278:false 1632421214573:false 1632421239500:false 1632491773213:false 1632521010878:false 1632612248621:false 1632763168185:false 1632770715550:false 1632789599152:false 1632856287012:false 1632866287695:false 1632919422143:false 1632945960455:false 1632948886775:false 1632951839103:false 1632954526814:false 1633037202485:false 1633102909627:false 1633114575804:false 1633355805941:false 1633378690300:false 1633380095416:false 1633383358825:false 1633383437966:false 1633391144528:false 1633446085959:false 1633463118767:false 1633557843562:false 1633557887887:false 1633558260666:false 1633618635957:false 1633634858220:false 1633694012222:false 1633696349662:false 1634060627104:false 1634138959038:false 1634159897075:false 1634310727741:false 1634316030243:false 1634568174930:false 1634588884759:false 1634588969755:false 1634600738903:false 1634600747387:false 1634600831348:false 1634601064433:false 1634602137404:false 1634602407869:false 1634744815824:false 1634745565606:false 1634747605943:false 1634766261961:false 1634908982721:false 1634918507135:false 1634918545905:false 1634924635718:false 1634924813895:false 1634935741651:false 1634936288768:false 1634936408072:false 1634968488205:false 1635215100875:false 1635223955116:false 1635223975907:false 1635276749735:false 1635276824418:false 1635276936189:false 1635277008994:false 1635304462604:false 1635304742616:false 1635358841791:false 1635369586323:false 1635369599439:false 1635427756566:false 1635435622631:false 1635526080143:false 1635557465843:false 1635618575008:false 1635798377520:false 1635798428881:false 1635798468595:false 1635798535558:false 1635799107958:false 1635799134592:false 1635799164605:false 1636034580283:false 1636034993961:false 1636043725958:false 1636043830277:false 1636044516164:false 1636048344567:false 1636048567632:false 1636065790385:false 1636065913045:false 1636151871792:false 1636253014655:false 1636402780561:false 1636412496018:false 1636412516592:false 1636474169465:false 1636475181749:false 1636480254025:false 1636481656134:false 1636481683166:false 1636481701299:false 1636483643013:false 1636494732494:false 1636496147948:false 1636510437615:false 1636511142660:false 1636512065350:false 1636518337880:false 1636565598054:false 1636571004444:false 1636586316672:false 1636607822780:false 1636665870874:false 1636672028957:false 1636763480813:false 1637018203642:false 1637096829362:false 1637160611267:false 1637160984668:false 1637166590163:false 1637167407403:false 1637171509576:false 1637186915412:false 1637245051101:false 1637259838529:false 1637274723366:false 1637600378100:false 1637627402600:false 1637630619763:false 1637699188094:false 1637760267680:false 1637760343646:false 1637761102562:false 1637772345237:false 1637778072029:false 1637778196429:false 1637795249079:false 1637795899701:false 1638212473273:false 1638233591605:false 1638297420654:false 1638566736230:false 1638567883425:false 1638587117274:false 1639075995044:false 1639077953798:false 1639161631084:false 1639177750140:false 1639255277711:false 1639328438438:false 1639423177766:false 1639423288757:false 1639435529433:false 1639700318237:false 1639700891266:false 1639748627330:false 1639748893957:false 1639753368798:false 1639754063605:false 1639766412059:false 1639778316346:false 1640010018455:false 1640034916007:false 1640092266450:false 1640096113762:false 1640096586678:false 1640114542691:false 1640129820112:false 1640189090685:false 1640192009463:false 1640210427699:false 1640223156925:false 1640223551631:false 1640224328605:false 1640289205715:false 1640294645408:false 1640295871208:false 1640359207299:false 1640719164155:false 1641185756329:false 1641229794613:false 1641235603296:false 1641245864993:false 1641314538008:false 1641401013540:false 1641401059203:false 1641428685632:false 1641491088576:false 1641497357536:false 1641497384299:false 1641734181493:false 1641756748928:false 1641761988421:false 1641779100975:false 1641825996216:false 1641830381487:false 1641859998558:false 1641870573905:false 1641951775932:false 1641951803461:false 1641952957847:false 1641953838922:false 1641955411200:false 1641957594741:false 1642021374605:false 1642021614853:false 1642030183128:false 1642032544072:false 1642040423639:false 1642085662173:false 1642108892541:false 1642109600089:false 1642128984762:false 1642129539461:false 1642142597272:false 1642145858412:false 1642188654892:false 1642189329380:false 1642281678134:false 1642435173579:false 1642439530956:false 1642449058088:false 1642463913684:false 1642523050817:false 1642546374157:false 1642556443385:false 1642556458222:false 1642556495199:false 1642564737560:false 1642566067731:false 1642566110874:false 1642568314697:false 1642568332696:false 1642614525851:false 1642621812626:false 1642622497413:false 1642622513112:false 1642622563508:false 1642622577984:false 1642622628140:false 1642622638046:false 1642622718480:false 1642622735194:false 1642630113899:false 1642630185085:false 1642630300224:false 1642634564106:false 1642634588724:false 1642636197156:false 1642699214713:false 1642701177856:false 1642720581217:false 1642723999534:false 1642724267340:false 1642724400897:false 1642724599349:false 1642724896205:false 1642725305788:false 1642725539919:false 1642776958789:false 1642781006590:false 1642868372663:false 1643033999010:false 1643034064320:false 1643037708652:false 1643044795726:false 1643069402862:false 1643080940311:false 1643119248367:false 1643122177470:false 1643131891194:false 1643142476747:false 1643157674316:false 1643157973577:false 1643158646418:false 1643160392454:false 1643232414227:false 1643256075030:false 1643293375909:false 1643293537097:false 1643315321568:false 1643318976939:false 1643319466067:false 1643323468040:false 1643329934750:false 1643331393902:false 1643390542662:false 1643397739808:false 1643402438287:false 1643504113666:false 1643524869404:false 1643644901792:false 1643655212094:false 1643661920374:false 1643726063821:false 1643738086883:false 1643750164633:false 1643750385245:false 1643750631316:false 1643752454677:false 1643752472815:false 1643824668882:false 1643825615871:false 1643827899016:false 1643833196218:false 1643948973158:false 1643950730291:false 1643951607017:false 1643954374954:false 1643994554103:false 1644011811475:false 1644020654311:false 1644034894417:false 1644037005698:false 1644099713531:false 1644117358352:false 1644117535967:false 1644283282511:false 1644290352753:false 1644332245307:false 1644344945655:false 1644348711901:false 1644349839555:false 1644350321002:false 1644374537814:false 1644374612600:false 1644385293403:false 1644386108039:false 1644435592477:false 1644435606944:false 1644442160088:false 1644608580862:false 1644608888838:false 1644617434084:false 1644640978645:false 1644874049871:false 1644927671274:false 1644944828231:false 1644945518067:false 1644945666235:false 1644945692960:false 1644950890730:false 1645027758882:false 1645120406996:false 1645134270437:false 1645147840042:false 1645148252710:false 1645157981119:false 1645190174707:false 1645218687763:false 1645420685346:false 1645454135050:false 1645459254414:false 1645468503572:false 1645478863405:false 1645540808112:false 1645685469287:false 1645686223380:false 1645731876381:false 1645732454401:false 1645817056576:false 1645817403904:false 1645817597318:false 1645827962293:false 1646067591046:false 1646068880219:false 1646081902245:false 1646082425923:false 1646140710584:false 1646168404372:false 1646198071238:false 1646238345290:false 1646312994012:false 1646325653676:false 1646326888855:false 1646327671627:false 1646401280404:false 1646684694837:false 1646688170056:false 1646688368316:false 1646753248701:false 1646761082879:false 1646761118026:false 1646766090016:false 1646766889762:false 1646772718898:false 1646773181877:false 1646783383076:false 1646787418337:false 1646837533637:false 1646839841354:false 1646840295136:false 1646843134067:false 1646844973302:false 1646849444827:false 1646855323310:false 1646856208158:false 1646871403300:false 1646926363594:false 1646926831250:false 1646945416796:false 1646950950810:false 1646979683689:false 1647038138548:false 1647040910761:false 1647047109918:false 1647283341048:false 1647288791598:false 1647292222442:false 1647372090708:false 1647373488824:false 1647383208263:false 1647532156412:false 1647536226502:false 1647537340405:false 1647541435577:false 1647625098499:false 1647650783404:false 1647884558485:false 1647885462657:false 1647885522015:false 1647885767633:false 1647890602034:false 1647891181674:false 1647960943615:false 1647981317629:false 1647986050267:false 1647986100625:false 1647986141940:false 1648056048791:false 1648056095797:false 1648071546299:false 1648074807156:false 1648144157230:false 1648147948435:false 1648149687467:false 1648150033649:false 1648154758524:false 1648155774319:false 1648156885949:false 1648158021078:false 1648158139663:false 1648233588539:false 1648234516924:false 1648237586302:false 1648255268454:false 1648572398104:false 1648580131029:false 1648582520922:false 1648583571213:false 1648590186915:false 1648651882184:false 1648656918630:false 1648657672860:false 1648658168936:false 1648661030115:false 1648662115629:false 1648665032725:false 1648680675591:false 1648733087941:false 1648733475807:false 1648759219120:false 1648761404027:false 1648830203453:false 1648840432634:false 1648841717720:false 1648847022185:false 1649105151351:false 1649162729781:false 1649162780641:false 1649181814038:false 1649183680352:false 1649189298721:false 1649201462940:false 1649201517519:false 1649201553767:false 1649270883495:false 1649346965902:false 1649356850586:false 1649358489039:false 1649362353920:false 1649431538457:false 1649439126871:false 1649439301848:false 1649450878230:false 1649637058468:false 1649640536038:false 1649641550459:false 1649656750531:false 1649691191281:false 1649698379630:false 1649711735632:false 1649714025791:false 1649744223933:false 1649788372822:false 1649872804367:false]] 
```

### Any possible solutions?

I'm not sure! I tried downgrading to a previous CLI version, and the issue with slowness does not exist there. Therefore this is definitely an issue introduced by 2.5.0. 

### Can you identify the location in the source code where the problem exists?

Nope! 

### If the bug is confirmed, would you be willing to submit a PR?

Maybe!

**Comment by rikinsk** [comment]
Hey thanks for reporting this. We are investigating this.

**Comment by osdiab** [comment]
Also the hasura console even for local development got insanely slow. Takes like 30 seconds to update permissions for a single table now.

**Comment by osdiab** [comment]
For reference I'm on an M1 Macbook, if that may be relevant to these issues.

**Comment by haf** [comment]
Impossible to use now with version 2.5. For medium large setups, the to just list what migrations are there is 1m 42s.

**Comment by ferm10n** [comment]
Possibly related, `hasura metadata export` is also really slow now too (250sec, after downgrading to 2.4.0 it's 2 seconds).
Also noticed it slams the CPU
![image](https://user-images.githubusercontent.com/7970987/163876301-4873ab7d-b605-4509-af22-8f9df1ac374d.png)

**Comment by osdiab** [comment]
@rikinsk any updates? Until this is fixed, myself, my team, and probably many other engineers will stop updating the Hasura CLI altogether. Totally destroys my ability to use Hasura as a developer.

**Comment by ferm10n** [comment]
1. hasura server version: 2.0.4
2. cli project version: 3 (This is the `config.yaml > version`, right?)
3. OS Info:
```
OS: Ubuntu 18.04.6 LTS x86_64 
Kernel: 5.4.0-107-generic
Shell: zsh 5.4.2 
CPU: Intel i7-7700 (8) @ 4.200GH 
GPU: Intel HD Graphics 630 
Memory: 6170MiB / 31999MiB
```

**Comment by efstathiosntonas** [comment]
1. 2.6.0 beta 1
2. hasura-cli 2.5.0, config v3
3. Macbook pro 2019 i7

No matter what I do, raw sql, edit permissions, edit column types etc, it's slow compared to older versions and I've also noticed cpu spikes while the operation is running.

**Comment by osdiab** [comment]
1. server version 2.3.0
2. cli project config version 3
3. OS info: M1 Max, 64GB RAM, Macbook Pro 16" 

<details>
```
DEBU global config file exists, verifying contents 
DEBU global config is pre-set to &cli.GlobalConfig{UUID:"2b9bf2be-59ef-488b-84d7-26887a1f2219", EnableTelemetry:true, ShowUpdateNotification:true, CLIEnvironment:"default"} 
DEBU global config: uuid: 2b9bf2be-59ef-488b-84d7-26887a1f2219 
DEBU global config: enableTelemetry: true         
DEBU global config: showUpdateNotification: true  
DEBU global config: cliEnvironment: default       
DEBU versions: cli: [v2.5.0] server: [v2.3.0]     
DEBU compatibility check: [true] cli and server does not match 
DEBU cannot get config information from server, this might be because config API is not enabled: error unmarshalling fetching server config 
DEBU graphql engine endpoint: http://localhost:8080 
DEBU graphql engine admin_secret: ‹×›             
DEBU server: uuid: 18ace845-2383-417f-87ba-b1895f9823cd 
DEBU skipping state migration, found IsStateCopyCompleted: true Migrations: map[default:map[1621321854518:false 1621895901253:false 1622687318514:false 1622694822270:false 1623046969028:false 1623047204684:false 1623047532523:false 1623047580211:false 1623047678608:false 1623047808432:false 1623047824606:false 1623048641322:false 1623652914601:false 1623653535173:false 1623736743901:false 1624040054599:false 1624040100637:false 1624040110671:false 1624040175403:false 1624040181891:false 1624040190124:false 1624398734426:false 1624398749355:false 1624398763720:false 1624398778865:false 1624398813072:false 1624398822067:false 1624398860480:false 1624398872408:false 1624398887619:false 1624399087572:false 1624409801023:false 1624409806939:false 1624410001811:false 1628528739023:false 1628531865627:false 1628559556739:false 1628606350100:false 1628606388115:false 1628606433351:false 1628969095494:false 1628971882070:false 1628971893365:false 1628971921784:false 1628986999965:false 1628987018031:false 1629307011160:false 1629307110421:false 1629317470549:false 1629401416824:false 1629401438049:false 1629401446597:false 1629401474443:false 1629401478933:false 1629421981748:false 1629421993120:false 1629422056801:false 1629422074237:false 1629422097904:false 1629422118880:false 1629484253653:false 1629505785809:false 1629508421483:false 1629509282398:false 1629774301565:false 1629853226068:false 1629854728874:false 1629913717938:false 1629920716397:false 1629930037159:false 1629942024713:false 1629999557669:false 1629999742816:false 1629999800062:false 1630003881557:false 1631832328479:false 1631832342738:false 1632022303815:false 1632022521388:false 1632023381996:false 1632023435272:false 1632107534942:false 1632118320997:false 1632740951907:false 1632745282544:false 1632745339226:false 1632814788365:false 1632962730581:false 1632980205132:false 1632988166042:false 1633411761803:false 1633411781163:false 1633412092770:false 1633412100978:false 1633412109166:false 1633412141161:false 1633412155870:false 1633413998894:false 1633415215265:false 1633419685780:false 1633420153259:false 1633420207113:false 1633420674988:false 1633421904710:false 1633421998296:false 1633422014806:false 1633424134006:false 1633428169192:false 1633431098271:false 1633506091853:false 1633520849776:false 1633521148463:false 1633521182584:false 1633521410234:false 1633521418778:false 1633521430668:false 1633556906529:false 1633606592101:false 1633614053079:false 1633674658373:false 1633674694137:false 1633674742216:false 1633952809499:false 1633955570343:false 1633955789618:false 1633959086592:false 1633960313467:false 1633993515582:false 1634109670480:false 1634122035127:false 1634122743902:false 1634199927704:false 1634301805553:false 1634548216754:false 1634548824614:false 1634549841785:false 1634551738034:false 1634552569148:false 1634639423187:false 1634639458296:false 1634754376624:false 1634858420761:false 1635159051150:false 1635487999030:false 1635731452509:false 1635756734571:false 1635757228682:false 1635879308550:false 1635879321615:false 1636108774360:false 1636545597378:false 1636560411301:false 1636560432170:false 1636560476169:false 1636561623011:false 1636675150200:false 1637069511850:false 1637244793284:false 1637297439081:false 1637568504971:false 1638353066850:false 1638368366118:false 1638455663951:false 1638801362971:false 1638843812097:false 1638847539730:false 1638885847896:false 1638908866573:false 1638961290140:false 1639150314391:false 1639150352468:false 1639152480223:false 1639373805292:false 1640265361476:false 1640266361176:false 1640266395946:false 1640322632181:false 1640344438351:false 1640344462336:false 1640602146295:false 1640674348126:false 1640778103776:false 1640875053362:false 1641219087682:false 1641284358300:false 1641298923343:false 1641299103338:false 1641299367805:false 1641299380747:false 1641316072488:false 1641360663666:false 1641360722024:false 1641360767554:false 1641360785507:false 1641360900347:false 1641444668991:false 1641474544759:false 1641474591717:false 1641474896361:false 1641474946227:false 1641474951703:false 1641475291805:false 1641475337795:false 1641476387833:false 1641477816737:false 1641660208692:false 1641680562821:false 1641796313999:false 1641796765040:false 1641797650023:false 1641812800307:false 1641894740903:false 1641918929751:false 1641932460021:false 1641946188519:false 1641947379106:false 1641953397600:false 1641954644987:false 1641989390759:false 1642015154884:false 1642015164969:false 1642031363705:false 1642073697927:false 1642080217365:false 1642137075295:false 1642141808547:false 1642417072256:false 1642512377555:false 1642587991037:false 1642681600699:false 1642681600700:false 1642772014099:false 1642880895110:false 1642995794311:false 1643018104544:false 1643018215734:false 1643112773145:false 1643114543230:false 1643114571658:false 1643118578807:false 1643191779355:false 1643193724911:false 1643201094280:false 1643203297173:false 1643234943769:false 1643237654949:false 1643276323590:false 1643298196166:false 1643549830488:false 1643713933542:false 1643727214805:false 1643985325024:false 1644228530237:false 1644243968969:false 1644322681561:false 1644327675313:false 1644365832247:false 1644368052703:false 1644416769966:false 1644461316892:false 1644483054572:false 1644486899905:false 1644494417140:false 1644497729984:false 1644580509141:false 1644743904368:false 1644821825954:false 1645010231667:false 1645014553365:false 1645104347866:false 1645104596712:false 1645115442592:false 1645165575686:false 1645566884224:false 1645711100288:false 1645765262478:false 1645774392959:false 1645774392960:false 1645779992719:false 1646371141913:false 1646705574334:false 1646705599729:false 1646841195014:false 1646841760915:false 1646842051760:false 1646873370004:false 1646905418585:false 1646959884832:false 1646965806083:false 1646971116942:false 1647102349208:false 1647236581721:false 1647236581722:false 1647236661580:false 1647322510715:false 1647335633557:false 1647339756514:false 1647343234852:false 1647348665466:false 1647349941814:false 1647419104102:false 1647950824289:false 1648221207227:false 1648552136404:false 1648566257245:false 1648566355868:false 1648566374373:false 1648591894822:false 1648663745967:false 1648722376450:false 1648731166796:false 1648760077321:false 1649075845878:false 1649172679915:false 1649178677615:false 1649240658430:false 1649243168169:false 1649243168170:false 1649243168171:false 1649336285834:false 1649683864132:false 1649683899331:false 1649724459468:false 1649726268253:false 1649730790908:false 1649763175589:false 1649770782048:false 1649867708707:false 1649919559375:false 1649943512849:false 1649944688615:false 1650002450609:false 1650003189963:false 1650003858319:false 1650004219831:false 1650004402107:false 1650005210131:false 1650005682873:false 1650005806548:false 1650006165594:false 1650015875605:false 1650252759014:false 1650274293774:false 1650331165667:false 1650331165668:false 1650351900581:false 1650416397609:false]] 
INFO nothing to apply on database default                                                                                                                      
DEBU telemetry: beamed                             fields.time=1.2231575s isError=false
```

paused on the line `DEBU server: uuid: 18ace845-2383-417f-87ba-b1895f9823cd` for a long time

</details>

**Comment by aaronhayes** [comment]
1. server version 2.5.0
2. cli project version 3 
3. OS Info: MacOS 12.3.1 Macbook Pro Core i9 16GB Ram. 

Using `v2.4.0` cli is fast.

**Comment by jflambert** [comment]
@gurukulkarni why reopen??? the fix is out with 2.5.1

**Comment by aaronhayes** [comment]
I can confirm issue fixed in 2.5.1. No need to reopen.
